nfts/src/tests/extrinsics.rs

```diff
@@ -9,375 +9,665 @@ fn origin(account: u64) -> mock::Origin {
+				);
+				assert_eq!(Balances::free_balance(BOB), 100);
+			})
+	}
+
+	#[test]
+	fn create_error_locked_serie() {
+		ExtBuilder::default().caps(vec![(ALICE, 100)]).build().execute_with(|| {
+			// State checks
+			assert_eq!(NFTs::nft_id_generator(), 0);
+			assert_eq!(NFTs::series_id_generator(), 0);
+
+			// Initial state
+			let alice: mock::Origin = origin(ALICE);
+
+			let series_id = Some(vec![51]);
+			assert_ok!(NFTs::create(alice.clone(), vec![50], series_id.clone()));
+			NFTs::finish_series(alice.clone(), series_id.clone().unwrap()).unwrap();
+
+			// Should fail and storage should remain empty
+			assert_noop!(
+				NFTs::create(alice, vec![1], series_id.clone()),
+				Error::<Test>::CannotCreateNFTsWithCompletedSeries
+			);
 		})
 	}
+}
+
+mod transfer {
+	use super::*;
 
 	#[test]
-	fn nft_not_found() {
-		ExtBuilder::new_build(vec![]).execute_with(|| {
-			let ok = NFTs::delegate(origin(ALICE), INVALID_NFT_ID, None);
-			assert_noop!(ok, Error::<Test>::NFTNotFound);
+	fn transfer_ok() {
+		ExtBuilder::default().caps(vec![(ALICE, 1000)]).build().execute_with(|| {
+			// Initial state
+			let alice: mock::Origin = origin(ALICE);
+			let series_id = vec![2];
+			let nft_id =
+				<NFTs as NFTTrait>::create_nft(ALICE, vec![1], Some(series_id.clone())).unwrap();
+			NFTs::finish_series(alice.clone(), series_id).unwrap();
+			let nft = NFTs::data(nft_id).unwrap();
+
+			// NFT owner and creator check
+			assert_eq!(nft.owner, ALICE);
+			assert_eq!(nft.creator, ALICE);
+
+			// Transfer nft ownership from ALICE to BOB
+			assert_ok!(NFTs::transfer(alice.clone(), nft_id, BOB));
+
+			// Events checks
+			assert_eq!(
+				System::events().last().unwrap().event,
+				Event::NFTs(NFTsEvent::NFTTransferred { nft_id, old_owner: ALICE, new_owner: BOB })
+			);
+
+			// Final state checks
+			let nft = NFTs::data(nft_id).unwrap();
+			assert_eq!(nft.owner, BOB);
+			assert_eq!(nft.creator, ALICE);
 		})
 	}
 
 	#[test]
-	fn not_the_nft_owner() {
+	fn transfer_error_unknown_nft() {
 		ExtBuilder::new_build(vec![(ALICE, 100)]).execute_with(|| {
-			let nft_id = NFTs::create_nft(ALICE, vec![0], None).unwrap();
+			// Initial state
+			let alice: mock::Origin = origin(ALICE);
 
-			let ok = NFTs::delegate(origin(BOB), nft_id, None);
-			assert_noop!(ok, Error::<Test>::NotTheNFTOwner);
+			// Try to transfer with an unknown nft id
+			// Should fail and storage should remain empty
+			assert_noop!(
+				NFTs::transfer(alice.clone(), INVALID_NFT_ID, BOB),
+				Error::<Test>::NFTNotFound
+			);
 		})
 	}
 
 	#[test]
-	fn cannot_delegate_nfts_listed_for_sale() {
+	fn transfer_error_uncompleted_serie() {
 		ExtBuilder::new_build(vec![(ALICE, 100)]).execute_with(|| {
-			let nft_id = NFTs::create_nft(ALICE, vec![0], None).unwrap();
-			assert_ok!(NFTs::set_listed_for_sale(nft_id, true));
+			// Initial state
+			let alice: mock::Origin = origin(ALICE);
+			let nft_id = <NFTs as NFTTrait>::create_nft(ALICE, vec![0], None).unwrap();
```

Comment:
You are constantly creating the same nft in all other tests. This is a good case for having this NFT already in the genesis config

Reply:
Sometimes tests don't need nft, ex: NFTNotFound, so with this I would instantiate it for nothing, I could do special builders for recurrent cases ?

Reply:
The tests that don't need it will not use it. Quite simple.

Reply:
Done